src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +282,196 @@ def new_component(
+    return "valid"
+
+
+def get_port_input():
+    # Gather inputs to use as context for the port template
+    defaults = {
+        "port_name": "ExamplePort",
+        "short_description": "Example usage of port",
+        "dir_name": "example_directory",
+        "arg_number": 1,
+    }
+    valid_name = False
+    valid_dir = False
+    while not valid_name:
+        port_name = input("Port Name [{}]: ".format(defaults["port_name"]))
+        char = is_valid_name(port_name)
+        if char != "valid":
+            print("'" + char + "' is not a valid character. Enter a new port name:")
+        else:
+            valid_name = True
+    short_description = input(
+        "Short Description [{}]: ".format(defaults["short_description"])
+    )
+    while not valid_dir:
+        dir_name = input("Directory Name [{}]: ".format(defaults["dir_name"]))
+        char = is_valid_name(dir_name)
+        if char != "valid":
+            print(
+                "'" + char + "' is not a valid character. Enter a new directory name:"
+            )
+        else:
+            valid_dir = True
+    string_arg_number = input(
+        "Number of arguments [{}]: ".format(defaults["arg_number"])
+    )
+    if string_arg_number == "":
+        arg_number = 1
+    elif not string_arg_number.isnumeric():
+        print("[ERROR] You have not entered a valid number")
+    else:
+        arg_number = int(string_arg_number)
+    values = {
+        "port_name": port_name,
+        "short_description": short_description,
+        "dir_name": dir_name,
+        "arg_number": arg_number,
+    }
+
+    # Fill in blank values with defaults
+    for key in values:
+        if values[key] == "":
+            values[key] = defaults[key]
+    return values
+
+
+def make_namespace(deployment, cwd):
+    # Form the namespace from the path to the deployment
+    namespace_path = cwd.relative_to(deployment)
+    deployment_dir = deployment.name
+    whole_path = Path(deployment_dir, namespace_path)
+    namespace = whole_path.parent
+    namespace_formatted = str(namespace).replace(os.path.sep, "::")
+    return namespace_formatted
+
+
+def new_port(cwd: Path, deployment: Path, settings: Dict[str, str]):
+    """ Uses cookiecutter for making new ports """
+    try:
+        print("[WARNING] **** fprime-util new is prototype functionality ****")
+        proj_root = None
+        try:
+            proj_root = Path(settings.get("project_root", None))
+        except (ValueError, TypeError):
+            print("[WARNING] No found project root.")
+
+        PATH = os.path.dirname(os.path.abspath(__file__))
+        TEMPLATE_ENVIRONMENT = Environment(
+            autoescape=False,
+            loader=FileSystemLoader(os.path.join(PATH, "../cookiecutter_templates")),
+            trim_blocks=False,
+        )
+        params = get_port_input()
+        if (
+            settings.get("framework_path") is not None
+            and settings["framework_path"] != "native"
```

Comment:
Framework path will always be a path, right? Is there a situation where it could get set to "native"?

Reply:
I think I may actually be able to get rid of this block because framework path is only used in the line that gets the schema in the port_template.xml